src/explore-education-statistics-admin/src/pages/release/data/components/DataFilesTable.tsx

```diff
@@ -0,0 +1,239 @@
+import Link from '@admin/components/Link';
+import DataFileDetailsTable from '@admin/pages/release/data/components/DataFileDetailsTable';
+import DataUploadCancelButton from '@admin/pages/release/data/components/DataUploadCancelButton';
+import ImporterStatus, {
+  terminalImportStatuses,
+} from '@admin/pages/release/data/components/ImporterStatus';
+import {
+  releaseApiDataSetDetailsRoute,
+  releaseDataFileReplaceRoute,
+  ReleaseDataFileReplaceRouteParams,
+  releaseDataFileRoute,
+  ReleaseDataFileRouteParams,
+  ReleaseDataSetRouteParams,
+} from '@admin/routes/releaseRoutes';
+import {
+  DataFile,
+  DataFileImportStatus,
+} from '@admin/services/releaseDataFileService';
+import ButtonGroup from '@common/components/ButtonGroup';
+import ButtonText from '@common/components/ButtonText';
+import Modal from '@common/components/Modal';
+import ReorderableList from '@common/components/ReorderableList';
+import reorder from '@common/utils/reorder';
+import React, { useEffect, useState } from 'react';
+import { generatePath } from 'react-router';
+import styles from './DataFilesTable.module.scss';
+
+interface Props {
+  dataFiles: DataFile[];
+  publicationId: string;
+  releaseId: string;
+  canUpdateRelease?: boolean;
+  isReordering: boolean;
+  onCancelReordering: () => void;
+  onConfirmReordering: (nextSeries: DataFile[]) => void;
+  onDeleteFile: (dataFile: DataFile) => Promise<void>;
+  onStatusChange: (
+    dataFile: DataFile,
+    { totalRows, status }: DataFileImportStatus,
+  ) => Promise<void>;
+}
+
+export default function DataFilesTable({
+  dataFiles: initialDataFiles,
+  publicationId,
+  releaseId,
+  canUpdateRelease,
+  isReordering,
+  onCancelReordering,
+  onConfirmReordering,
+  onDeleteFile,
+  onStatusChange,
+}: Props) {
+  const [dataFiles, setDataFiles] = useState(initialDataFiles);
+
+  useEffect(() => {
+    setDataFiles(initialDataFiles);
+  }, [initialDataFiles]);
+
+  if (isReordering) {
+    return (
+      <ReorderableList
+        heading="Reorder data files"
+        id="dataFiles"
+        list={dataFiles.map(({ id, title }) => ({
+          id,
+          label: title,
+        }))}
+        onCancel={() => {
+          setDataFiles(initialDataFiles);
+          onCancelReordering();
+        }}
+        onConfirm={() => onConfirmReordering(dataFiles)}
+        onMoveItem={({ prevIndex, nextIndex }) => {
+          const reordered = reorder(dataFiles, prevIndex, nextIndex);
+          setDataFiles(reordered);
+        }}
+        onReverse={() => {
+          setDataFiles(dataFiles.toReversed());
+        }}
+      />
+    );
+  }
+
+  return (
+    <table className={styles.table} data-testid="Data files table">
+      <thead>
+        <tr>
+          <th scope="col">Title</th>
+          <th scope="col">Size</th>
+          <th scope="col">Status</th>
+          <th scope="col">Actions</th>
```

Comment:
Seems like we're missing a bunch of columns that appear in the ticket and acceptance criteria? Is this on purpose? If so, the ticket needs to be updated. 

If not, looks like we need to add those in. We'll need to update the tests accordingly as well!

Reply:
With the addition of the "View details" modal, we've opted to remove some columns to make the rows more concise at first glance wit the added option to see more information. [Cam and I have discussed briefly here](https://dfedigital.atlassian.net/browse/EES-5544?focusedCommentId=100335) in the ticket comments

Reply:
The ticket has been updated accordingly now